external_server/server.py

```
@@ -0,0 +1,879 @@
+
+        self._known_devices.clear()
+        self._event_queue.clear()
+
+    def _clear_modules(self) -> None:
+        """Stop the threads for each server module and destroy the API adapters."""
+        for module in self._modules.values():
+            module.thread.stop()
+        for module in self._modules.values():
+            module.thread.wait_for_join()
+            code = module.api.destroy()
+            if not code == GeneralErrorCode.OK:
```

Ruff

Use `code != GeneralErrorCode.OK` instead of `not code == GeneralErrorCode.OK`

Replace with `!=` operator

(SIM201)